packages/legend-application-studio/src/stores/editor/sidebar-state/end-to-end-workflow/GlobalEndToEndFlowState.ts

```diff
@@ -0,0 +1,598 @@
+      connectionValueState: observable,
+      databaseBuilderWizardState: observable,
+      packageableConnection: observable,
+      databaseModelBuilderState: observable,
+      isGeneratingDatabaseGrammarCode: observable,
+      createConnection: computed,
+      setActiveStep: action,
+      setDatabaseGrammarCode: action,
+      setTargetConnectionPath: action,
+      setRuntimeGrammarCode: action,
+      setTargetRuntimePath: action,
+      setFinalGrammarCode: action,
+      buildDatabaseModelBuilderState: action,
+      setCompileError: action,
+      reset: action,
+      generateDatabaseGrammarCode: flow,
+      updateGraphWithModels: flow,
+      buildDatabase: flow,
+      updateRuntime: flow,
+      query: flow,
+      compileDatabaseGrammarCode: flow,
+      compile: flow,
+    });
+    this.activeStep = 0;
+    this.isValid = true;
+    this.globalEndToEndWorkflowState = globalEndToEndWorkflowState;
+    this.queryConnectionEndToEndWorkflowEditorState =
+      new QueryConnectionEndToEndWorkflowEditorState(
+        this.globalEndToEndWorkflowState.editorStore,
+      );
+    this.connectionValueState = new RelationalDatabaseConnectionValueState(
+      this.globalEndToEndWorkflowState.editorStore,
+      observe_RelationalDatabaseConnection(
+        this.createConnection,
+        this.globalEndToEndWorkflowState.editorStore.changeDetectionState
+          .observerContext,
+      ),
+    );
+    this.graph =
+      this.globalEndToEndWorkflowState.editorStore.graphManagerState.createNewGraph();
+  }
+
+  setActiveStep(step: number): void {
+    this.activeStep = step;
+  }
+
+  setTargetConnectionPath(val: string): void {
+    this.targetConnectionPath = val;
+  }
+
+  setTargetRuntimePath(val: string): void {
+    this.targetRuntimePath = val;
+  }
+
+  setRuntimeGrammarCode(val: string): void {
+    this.runtimeGrammarCode = val;
+  }
+
+  setDatabaseGrammarCode(code: string): void {
+    this.databaseGrammarCode = code;
+  }
+
+  setFinalGrammarCode(val: string): void {
+    this.finalGrammarCode = val;
+  }
+
+  buildDatabaseBuilderWizardState(): void {
+    this.databaseBuilderWizardState = new DatabaseBuilderWizardState(
+      this.globalEndToEndWorkflowState.editorStore,
+      this.connectionValueState.connection,
+      false,
+    );
+  }
+
+  setCompileError(err: ParserError | CompilationError | undefined): void {
+    this.compileError = err;
+  }
+
+  buildDatabaseModelBuilderState(): void {
+    this.databaseModelBuilderState = new DatabaseModelBuilderState(
+      this.globalEndToEndWorkflowState.editorStore,
+      guaranteeNonNullable(
+        this.databaseBuilderWizardState,
+        'databaseBuilderState must exist before building model from database',
+      ).schemaExplorerState.database,
+      false,
+    );
+    this.databaseModelBuilderState.assistanceLightGraph = this.graph;
+  }
+
+  *buildDatabase(): GeneratorFn<void> {
+    const databaseBuilderWizardState = guaranteeNonNullable(
```

Comment:
are we catching any of these errors? we risk going throwing errors in the UI here right?